### PR TITLE
Optimize HTTPHdr conversion of HTTP/2 to HTTP/1.1

### PR DESCRIPTION
**Pull request by masaori335**
- Avoid memcpy & parse for setting URL
- Avoid unnecessary MIMEField search by calling `field_delete(const char *name, int name_length)`

### CODE REVIEW

File: proxy/http2/HTTP2.cc [hunk 1]
```diff
@@ -406,88 +406,76 @@ http2_parse_window_update(IOVec iov, uint32_t &size)
...
   if (http_hdr_type_get(headers->m_http) == HTTP_TYPE_REQUEST) {
-    const char *scheme, *authority, *path;
-    int scheme_len, authority_len, path_len;
+    // :scheme
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_SCHEME, HTTP2_LEN_SCHEME); field != nullptr && field->value_is_valid()) {
+      int scheme_len;
+      const char *scheme = field->value_get(&scheme_len);
+
+      int scheme_wks_idx = hdrtoken_tokenize(scheme, scheme_len);
+      url_scheme_set(headers->m_heap, headers->m_http->u.req.m_url_impl, scheme, scheme_wks_idx, scheme_len, true);
```
Comment (maskit) [inline proxy/http2/HTTP2.cc]: Does this work even if the scheme is not wks? Or can we ignore the case?

Comment (masaori335) [reply proxy/http2/HTTP2.cc]: IIUC, `url_scheme_set()` copy `scheme` when `bool copy_string` flag (the last arg) is `true` even if `scheme_wks_idx` is `-1`.

**Comment by maskit** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
```
Running Test http2:......F. Failed
```
I'll dig this failure.

**Comment by masaori335** [comment]
The failure comes from treating the `path`. `url_print()` add `/`, so `/` needs to be removed.
Prior to this PR, `url_parse()` removed it.
```
DEBUG: <URL.cc:1607 (url_describe)> (http)     PATH: "huge_resp_hdrs", PATH_LEN: 14,
```
vs
```
DEBUG: <URL.cc:1607 (url_describe)> (http)     PATH: "/huge_resp_hdrs", PATH_LEN: 15,
```

File: proxy/http2/HTTP2.cc [hunk 2]
```diff
@@ -406,88 +406,82 @@ http2_parse_window_update(IOVec iov, uint32_t &size)
...
 
-    if ((field = headers->field_find(HTTP2_VALUE_PATH, HTTP2_LEN_PATH)) != nullptr && field->value_is_valid()) {
-      path = field->value_get(&path_len);
+    // :path
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_PATH, HTTP2_LEN_PATH); field != nullptr && field->value_is_valid()) {
+      int path_len;
+      const char *path = field->value_get(&path_len);
+
+      // cut first '/' if there, because `url_print()` add '/' before printing path
+      if (path_len >= 1 && path[0] == '/') {
```
Comment (maskit) [inline proxy/http2/HTTP2.cc]: Is this the same as what `url_parse` does? What if there are multiple slashes?

Comment (masaori335) [reply proxy/http2/HTTP2.cc]: I had the same question for multiple slashes and tried. If my experiment is correct, the only first `/` is removed.

Comment (maskit) [reply proxy/http2/HTTP2.cc]: Ok, I'm fine fine with this as long as the behavior is the same.

**Pull request merged**